ext/ext_skel.php

```diff
@@ -31,23 +31,86 @@ function error($message) {
 /* {{{ print_help
  */
 function print_help() {
-	printf('php ext_skel.php --ext <name> [--experimental] [--author <name>]%s', PHP_EOL);
-	printf('                 [--dir <path>] [--std] [--onlyunix]%s', PHP_EOL);
-	printf('                 [--onlywindows] [--help]%1$s%1$s', PHP_EOL);
-	printf('  --ext <name>		The name of the extension defined as <name>%s', PHP_EOL);
-	printf('  --experimental	Passed if this extension is experimental, this creates%s', PHP_EOL);
-	printf('                        the EXPERIMENTAL file in the root of the extension%s', PHP_EOL);
-	printf('  --author <name>       Your name, this is used if --std is passed and%s', PHP_EOL);
-	printf('                        for the CREDITS file%s', PHP_EOL);
-	printf('  --dir <path>		Path to the directory for where extension should be%s', PHP_EOL);
-	printf('                        created. Defaults to the directory of where this script%s', PHP_EOL);
-	printf(' 			lives%s', PHP_EOL);
-	printf('  --std			If passed, the standard header used%s', PHP_EOL);
-	printf(' 			in extensions that is included in the core, will be used%s', PHP_EOL);
-	printf('  --onlyunix		Only generate configure scripts for Unix%s', PHP_EOL);
-	printf('  --onlywindows		Only generate configure scripts for Windows%s', PHP_EOL);
-	printf('  --help                This help%s', PHP_EOL);
+    if (PHP_OS_FAMILY != 'Windows') {
+        $file_prefix = './';
+        $make_prefix = '';
+    } else {
+        $file_prefix = '';
+        $make_prefix = 'n';
+    }
 
+    echo <<<HELP
+WHAT IT IS
+
+  It's a tool for automatically creating the basic framework for a PHP extension.
+
+HOW TO USE IT
+
+  Very simple. First, change to the ext/ directory of the PHP sources. Then run
+  the following
+
+    php ext_skel.php --ext extension_name
+
+  and everything you need will be placed in directory ext/extension_name.
+
+  If you don't need to test the existence of any external header files,
+  libraries or functions in them, the extension is ready to be compiled in PHP.
+  To compile the extension in the PHP, run the following:
```

Comment:
This should be the alternative, by default you don't really want to rebuild php or build static, put phpize section first please.

Reply:
Refactored. Should the `print_success()` also be refactored? Because there it says to continue with the static build into the php:

```
printf('%1$sSuccess. The extension is now ready to be compiled into PHP. To do so, use the%s', PHP_EOL);
	printf('following steps:%1$s%1$s', PHP_EOL);
	printf('cd /path/to/php-src%s', PHP_EOL);
	printf('%sbuildconf%s', $file_prefix, PHP_EOL);
	printf('%sconfigure --enable-%s%s', $file_prefix, $options['ext'], PHP_EOL);
	printf('%smake%2$s%2$s', $make_prefix, PHP_EOL);
	printf('Don\'t forget to run tests once the compilation is done:%s', PHP_EOL);
	printf('%smake test TESTS=ext/%s/tests%3$s%3$s', $make_prefix, $options['ext'], PHP_EOL);
	printf('Thank you for using PHP!%s', PHP_EOL);
```

Reply:
I think it should as well Peter, at least for consistency reasons

Reply:
Added 👍